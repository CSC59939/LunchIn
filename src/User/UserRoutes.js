import React, { Component } from "react";
import { Redirect, Route } from "react-router-dom";
import UserHome from "./UserHome/UserHome.js";
import UserProfile from './UserHome/UserProfile/UserProfile.js';

class UserRoutes extends Component {
  render() {
    return (
      <div style={{ width: "100%", height: "100%" }}>
        <Route
          exact
          path={this.props.match.path}
          render={() => {
            return <Redirect to="/user/home" />;
          }}
        />
        <Route
          exact
          path={`${this.props.match.path}/home`}
          component={UserHome}
        />

        <Route
          exact
          path={this.props.match.path}
          render={() => {
            return <Redirect to="/user/profile" />;
          }}
        />
        <Route
          exact
          path={`${this.props.match.path}/profile`}
          component={UserProfile}
        />
      </div>
    );
  }
}
export default UserRoutes;
