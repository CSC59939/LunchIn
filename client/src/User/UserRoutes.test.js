import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import UserRoutes from "./UserRoutes";

describe("UserRoutes Component", () => {
  it("Should render correctly with correct path", () => {
    const match = { path: "/user/home" };
    const userRoutesInstance = shallow(<UserRoutes match={match} />);
    expect(userRoutesInstance).toMatchSnapshot();
  });
});
