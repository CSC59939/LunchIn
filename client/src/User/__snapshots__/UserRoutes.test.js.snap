// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRoutes Component Should render correctly with correct path 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserRoutes
    match={
      Object {
        "path": "/user/home",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          exact={true}
          path="/user/home"
          render={[Function]}
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/user/home/home"
        />,
        <Route
          exact={true}
          path="/user/home"
          render={[Function]}
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/user/home/profile"
        />,
        <Route
          exact={true}
          path="/user/home"
          render={[Function]}
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/user/home/zipcode"
        />,
      ],
      "style": Object {
        "height": "100%",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/user/home",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/user/home/home",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/user/home",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/user/home/profile",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/user/home",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/user/home/zipcode",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/user/home"
            render={[Function]}
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/user/home/home"
          />,
          <Route
            exact={true}
            path="/user/home"
            render={[Function]}
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/user/home/profile"
          />,
          <Route
            exact={true}
            path="/user/home"
            render={[Function]}
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/user/home/zipcode"
          />,
        ],
        "style": Object {
          "height": "100%",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/user/home",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/user/home/home",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/user/home",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/user/home/profile",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/user/home",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/user/home/zipcode",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
